### ISSUE TITLE
Testing Ionic 4 - Unauthorized on Ionic Pro

### ISSUE THREAD

**Issue by peterennis** [ionic-team/ionic-cli#3039]
**Description:**
Run through install process for new project.
Unauthorized. No prompt to sign in.

**Steps to Reproduce:**
Below

**Output:**
```
C:\ae\adaept.com>ionic info
[WARN] You are not in an Ionic project directory. Project context may be missing.
cli packages: (C:\Users\peter\AppData\Roaming\npm\node_modules)

   @ionic/cli-utils  : 2.0.0-rc.1
   ionic (Ionic CLI) : 4.0.0-rc.1

System:

   NodeJS : v8.9.4
   npm    : 5.6.0
   OS     : Windows 10


C:\ae\adaept.com>ionic config set -g features.project-angular true
[OK] features.project-angular set to true in ..\..\Users\peter\.ionic\config.json!

C:\ae\adaept.com>ionic start
[INFO] Every great app needs a name!

       We will use this name for the project's folder name and package name. You can change this at any time. To bypass
       this prompt next time, supply name, the first argument to ionic start.

? Project name: aeion4
[INFO] What type of project would you like to create?

       We recommend angular. To learn more about project types, see the CLI documentation[1]. To bypass this prompt next
       time, supply the --type option.

       [1]: https://ionicframework.com/docs/cli/starters.html

? Project type: angular
[INFO] Let's pick the perfect starter template!

       Starter templates are ready-to-go Ionic apps that come packed with everything you need to build your app. To
       bypass this prompt next time, supply template, the second argument to ionic start.

? Starter template: blank
√ Preparing directory .\aeion4 - done!
√ Downloading and extracting blank starter - done!
? Would you like to integrate your new app with Cordova to target native iOS and Android? Yes
> ionic integrations enable cordova --quiet
√ Downloading integration cordova - done!
√ Copying integrations files to project - done!
> npm i --save -E @ionic/cli-hooks
npm notice created a lockfile as package-lock.json. You should commit this file.
+ @ionic/cli-hooks@1.0.0-rc.1
added 3 packages in 1.204s
[OK] Integration cordova added!

Installing dependencies may take several minutes.

     *   IONIC  DEVAPP   *

 Speed up development with the Ionic DevApp, our fast, on-device testing mobile app

  -     Test on iOS and Android without Native SDKs
  -     LiveReload for instant style and JS updates

 -->    Install DevApp: https://bit.ly/ionic-dev-app    <--

────────────────────────────────────────────────────────────

> npm i
npm WARN deprecated nodemailer@2.7.2: All versions below 4.0.1 of Nodemailer are deprecated. See https://nodemailer.com/status/
npm WARN deprecated mailcomposer@4.0.1: This project is unmaintained
npm WARN deprecated socks@1.1.9: If using 2.x branch, please upgrade to at least 2.1.6 to avoid a serious bug with socket data flow and an import issue introduced in 2.1.0
npm WARN deprecated node-uuid@1.4.8: Use uuid module instead
npm WARN deprecated buildmail@4.0.1: This project is unmaintained

> uws@9.14.0 install C:\ae\adaept.com\aeion4\node_modules\uws
> node-gyp rebuild > build_log.txt 2>&1 || exit 0


> node-sass@4.8.3 install C:\ae\adaept.com\aeion4\node_modules\node-sass
> node scripts/install.js

Cached binary found at C:\Users\peter\AppData\Roaming\npm-cache\node-sass\4.8.3\win32-x64-57_binding.node

> uglifyjs-webpack-plugin@0.4.6 postinstall C:\ae\adaept.com\aeion4\node_modules\webpack\node_modules\uglifyjs-webpack-plugin
> node lib/post_install.js


> node-sass@4.8.3 postinstall C:\ae\adaept.com\aeion4\node_modules\node-sass
> node scripts/build.js

Binary found at C:\ae\adaept.com\aeion4\node_modules\node-sass\vendor\win32-x64-57\binding.node
Testing binary
Binary is fine
npm WARN @schematics/package-update@0.3.2 requires a peer of @angular-devkit/core@0.3.2 but none is installed. You must install peer dependencies yourself.
npm WARN @schematics/package-update@0.3.2 requires a peer of @angular-devkit/schematics@0.3.2 but none is installed. You must install peer dependencies yourself.
npm WARN @schematics/angular@0.3.2 requires a peer of @angular-devkit/core@0.3.2 but none is installed. You must install peer dependencies yourself.
npm WARN @schematics/angular@0.3.2 requires a peer of @angular-devkit/schematics@0.3.2 but none is installed. You must install peer dependencies yourself.
npm WARN ajv-keywords@3.1.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

added 1269 packages in 63.285s
> git init

     *   IONIC  PRO   *

 Supercharge your Ionic development with the Ionic Pro SDK

  -     Track runtime errors in real-time, back to your original TypeScript
  -    Push remote updates and skip the app store queue

 Learn more about Ionic Pro: https://ionicframework.com/pro

────────────────────────────────────────────────────────────

? Install the free Ionic Pro SDK and connect your app? Yes
> npm i --save -E @ionic/pro
> ionic link
? What would you like to do? Link an existing app on Ionic Pro
× Looking up your apps - failed!
Request: GET https://api.ionicjs.com/apps?page=1&page_size=100
Response: 401
Body:
{ link: null, message: 'Invalid Token', type: 'Unauthorized' }

C:\ae\adaept.com>
```

**My `ionic info`:**

```
C:\ae\adaept.com>cd aeion4

C:\ae\adaept.com\aeion4>ionic info
cli packages: (C:\Users\peter\AppData\Roaming\npm\node_modules)

   @ionic/cli-utils  : 2.0.0-rc.1
   ionic (Ionic CLI) : 4.0.0-rc.1

global packages:

   cordova (Cordova CLI) : 7.0.1

local packages:

   @angular-devkit/core       : 0.4.6
   @angular-devkit/schematics : 0.4.6
   @angular/cli               : 1.7.3
   @ionic/core                : 0.1.3
   Cordova Platforms          : none
   Ionic Framework            : @ionic/angular 0.0.2-29

System:

   NodeJS : v8.9.4
   npm    : 5.6.0
   OS     : Windows 10

Environment Variables:

   ANDROID_HOME : not set


C:\ae\adaept.com\aeion4>
```

**Other Information:**
WARN messages and unmaintained references on installation above.

**Comment by rdlabo** [comment]
Ionic Pro need login by `ionic signup` or `ionic login`. and login account can check  `ionic config get -g`.

ex)
```
{ tokens: { telemetry: '*****', user: '*****' },
  user: { id: 【user ID】, email: '【Email Address】' },
  git: { setup: true },
  telemetry: true,
  state:
   { lastCommand: '2018-03-27T13:34:11.871Z',
     lastNoResponseToUpdate: '2017-11-03T13:45:02.671Z',
     doctor: { ignored: [] } },
  addresses: {},
  npmClient: 'npm',
  features: { 'project-angular': true } }
```

probably, I think  you have not logged in.

**Comment by dwieeb** [comment]
@peterennis Very strange. Mine prompts for a login: 

<img width="464" alt="image" src="https://user-images.githubusercontent.com/236501/37986879-535bb554-31c2-11e8-9ec5-8d0c914c6501.png">

Can you send me the output of `ionic config get -g`?

**Comment by dwieeb** [comment]
Oops, I misread the original error. It says "Invalid Token". Can you think of any reason why that `tokens.user` value would be invalid?

**Comment by peterennis** [comment]
This issue: https://github.com/firebase/firebase-tools/issues/155
TLDR;
I created a new gmail account so the token may be a hangover from the previous one - drank
too many bits?

**Comment by peterennis** [comment]
Try again:

```
? Install the free Ionic Pro SDK and connect your app? Yes
> npm i --save -E @ionic/pro
> ionic link
? What would you like to do? Create a new app on Ionic Pro
? Please enter a name for your new app: aeion4
Request: POST https://api.ionicjs.com/apps
Response: 401
Body:
{ link: null, message: 'Invalid Token', type: 'Unauthorized' }

C:\ae\adaept.com>
```

**Comment by dwieeb** [comment]
@peterennis I can try to catch the error, but I'm guessing if you use `ionic login`, your token will be refreshed.

**Comment by peterennis** [comment]
Used ionic login to create a new account.
```
C:\ae\adaept.com\aeion4>ionic login
Log into your Ionic Pro account
If you don't have one yet, create yours by running: ionic signup
? Email: ...............@gmail.com
? Password: ***************
[OK] You are logged in!

C:\ae\adaept.com\aeion4>
```
